Add simple release process and info for custom helm repo.

thepetk opened the pull request:
### What does this PR do?:

* Moves the chatbot helm chart content one level up and updates all links. The previous path for the `Chart.yaml` was `charts/ai-software-templates/chatbot/0.1.0` and the new one is `charts/ai-software-templates/chatbot/.
* Adds a very simple release process for the initial release.
* The `chart-releaser.yaml` is replaced with the `release.yaml`. So everytime we cut a new release, e.g `v0.1.0` the helm packages are pinned.
* As we are going to clean the custom helm repository on the `ai-lab-helm-charts` repository, adds instructions on how someone can create a custom repository for their fork.

### Which issue(s) this PR fixes:

https://issues.redhat.com/browse/DEVAI-210

### PR acceptance criteria:

Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened and linked to this PR, if they are not in the PR scope due to various constraints.

- [ ] Tested and Verified

<!-- _I have verified that the changes were tested manually_ -->

- [ ] Documentation (READMEs, Product Docs, Blogs, Education Modules, etc.)

 <!-- _This includes READMEs, Product Docs, Blogs, Education Modules, etc._ -->

### How to test changes / Special notes to the reviewer:
I have tested the release.yaml here: https://github.com/thepetk/ai-lab-helm-charts/releases/tag/v0.1.0

FYI in case someone wants to import the updated helm repository in their cluster feel free to use my github pages here: https://thepetk.github.io/ai-lab-helm-charts/ as helm repo.

thepetk committed:
Remove 0.1.0 directory

thepetk committed:
Add info for custom helm repo

thepetk committed:
Add release process information

thepetk committed:
Replace release workflow

thepetk committed:
Specify chart version info

thepetk committed:
Remove relative path

Jdubrick commented on README.md:
```diff
@@ -19,3 +19,15 @@ The helm chart mainly uses the tekton pipelines under [rhdh-pipelines](https://g
+
+## Testing with a Custom Helm Repository
+
+In case you're interested to test your updates by importing the `ai-lab-helm-charts` fork as a custom helm chart repository, you can follow the instructions [here](./docs/SETUP_CUSTOM_HELM_REPO.md)
```
```suggestion
To test your updates by importing the `ai-lab-helm-charts` fork as a custom Helm chart repository, you can follow the instructions [here](./docs/SETUP_CUSTOM_HELM_REPO.md)
```

Jdubrick commented on README.md:
```diff
@@ -19,3 +19,15 @@ The helm chart mainly uses the tekton pipelines under [rhdh-pipelines](https://g
+
+In case you're interested to test your updates by importing the `ai-lab-helm-charts` fork as a custom helm chart repository, you can follow the instructions [here](./docs/SETUP_CUSTOM_HELM_REPO.md)
+
+## Release process
```
```suggestion
## Release Process
```

Jdubrick commented on README.md:
```diff
@@ -19,3 +19,15 @@ The helm chart mainly uses the tekton pipelines under [rhdh-pipelines](https://g
+
+The ai-lab-helm-charts are created on demand.
+
+- A `tag` should be created with the version of the release as name. ai-lab-helmcharts follows the v{major}.{minor}.{bugfix} format (e.g v0.1.0).
```
```suggestion
- A `tag` should be created with the version of the release as the name. `ai-lab-helm-charts` follows the v{major}.{minor}.{bugfix} format (e.g v0.1.0).
```

Jdubrick commented on README.md:
```diff
@@ -19,3 +19,15 @@ The helm chart mainly uses the tekton pipelines under [rhdh-pipelines](https://g
+The ai-lab-helm-charts are created on demand.
+
+- A `tag` should be created with the version of the release as name. ai-lab-helmcharts follows the v{major}.{minor}.{bugfix} format (e.g v0.1.0).
+- Before proceeding, make sure that all the `Chart.yaml` `version` fields have this tag as value. For example in case the tag is `v0.1.0` the `version` should be `0.1.0`.
```
```suggestion
- Before proceeding, make sure that all the `version` fields within `Chart.yaml` have this tag as the value. For example in the case where the tag is `v0.1.0`, the `version` should be `0.1.0`.
```

Jdubrick commented on README.md:
```diff
@@ -19,3 +19,15 @@ The helm chart mainly uses the tekton pipelines under [rhdh-pipelines](https://g
+
+- A `tag` should be created with the version of the release as name. ai-lab-helmcharts follows the v{major}.{minor}.{bugfix} format (e.g v0.1.0).
+- Before proceeding, make sure that all the `Chart.yaml` `version` fields have this tag as value. For example in case the tag is `v0.1.0` the `version` should be `0.1.0`.
+- After the new release is published, the updated helm packages will be pinned on the release.
```
```suggestion
- After the new release is published, the updated Helm packages will be pinned on the release.
```

Jdubrick commented on charts/ai-software-templates/chatbot/README.md:
```diff
@@ -9,7 +9,7 @@ The deployment flow, will create an application instance, a model server and a g
 - You need to have access to a cluster for each operation with OpenShift 4, like deploying and testing.
 - The Namespace that your application will run is already created in your cluster.
-- Your cluster should have [Openshift Pipelines Operator](https://www.redhat.com/en/technologies/cloud-computing/openshift/pipelines) installed and should be connected to your Github APP's webhook. In case your cluster is not configured yet, check the ["Pipelines Configuration Guide"](../../../../docs/PIPELINES_CONFIGURATION.md) for further instructions.
+- Your cluster should have [Openshift Pipelines Operator](https://www.redhat.com/en/technologies/cloud-computing/openshift/pipelines) installed and should be connected to your Github APP's webhook. In case your cluster is not configured yet, check the ["Pipelines Configuration Guide"](https://github.com/redhat-ai-dev/ai-lab-helm-charts/blob/main/docs/PIPELINES_CONFIGURATION.md) for further instru ...
```
```suggestion
- Your cluster should have [Openshift Pipelines Operator](https://www.redhat.com/en/technologies/cloud-computing/openshift/pipelines) installed and should be connected to your Github App's webhook. In case your cluster is not configured yet, check the ["Pipelines Configuration Guide"](https://github.com/redhat-ai-dev/ai-lab-helm-charts/blob/main/docs/PIPELINES_CONFIGURATION.md) for further instructions.
```

Jdubrick commented on docs/SETUP_CUSTOM_HELM_REPO.md:
```diff
@@ -0,0 +1,45 @@
+## Testing with a Custom Helm Repository
+
+In case you would like to setup a custom helm repository on your fork and be able to import and test the ai-lab-helm charts repo on your OCP cluster, you can follow the steps below:
```
```suggestion
In case you would like to setup a custom Helm repository on your fork and be able to import and test the `ai-lab-helm-charts` repo on your OCP cluster, you can follow the steps below:
```

Jdubrick commented on docs/SETUP_CUSTOM_HELM_REPO.md:
```diff
@@ -0,0 +1,45 @@
+
+### Add Helm Package and Index
+
+- Package the helm chart you want to test:
```
```suggestion
- Package the Helm chart you want to test:
```

Jdubrick commented on docs/SETUP_CUSTOM_HELM_REPO.md:
```diff
@@ -0,0 +1,45 @@
+
+- Push to your branch. A deployment will start after the push event.
+
+- On your OCP cluster you can create a custom helm repo by applying this yaml file:
```
```suggestion
- On your OCP cluster you can create a custom Helm repo by applying this yaml file:
```

Jdubrick commented on .github/workflows/release.yaml:
```diff
@@ -0,0 +1,40 @@
+on:
+  push:
+    tags:
+      - "v*"
```
For my understanding this workflow will run only when we create a tag starting with `v`?

thepetk replied:
yes exactly!

Jdubrick reviewed (commented):
Overall looks good, couple small comments and nits!

thepetk committed:
Address docs feedback

thepetk commented:
> Overall looks good, couple small comments and nits!

Thanks for the input :) I've addressed all comments as part of [200d198](https://github.com/redhat-ai-dev/ai-lab-helm-charts/pull/20/commits/200d198835fbb4e553f39603b2fe2b5a9fea3eb7)

thepetk committed:
Add pipelines route as webhook url

Jdubrick commented on docs/PIPELINES_CONFIGURATION.md:
```diff
@@ -4,7 +4,7 @@ The OpenShift Pipelines configuration is a requirement in order to support CI/CD
 1. Install the [Openshift Pipelines Operator](https://docs.redhat.com/en/documentation/openshift_container_platform/4.6/html/pipelines/installing-pipelines#installing-pipelines).
 
-2. Ensure that the `pipeline-as-code-controller` is up by getting its route:
+2. Ensure that the `pipeline-as-code-controller` is up by getting its route. Remember to add this route as the webhook url of your Github app:
```
nit
```suggestion
2. Ensure that the `pipeline-as-code-controller` is up by getting its route. Remember to add this route as the webhook url of your GitHub App:
```

thepetk replied:
Addressed in [f19e02d](https://github.com/redhat-ai-dev/ai-lab-helm-charts/pull/20/commits/f19e02d27583f22f67de4938fa5205423b8a7e3f)

thepetk committed:
Add correction to Github App naming

maysunfaisal commented on docs/PIPELINES_CONFIGURATION.md:
```diff
@@ -4,7 +4,7 @@ The OpenShift Pipelines configuration is a requirement in order to support CI/CD
 1. Install the [Openshift Pipelines Operator](https://docs.redhat.com/en/documentation/openshift_container_platform/4.6/html/pipelines/installing-pipelines#installing-pipelines).
 
-2. Ensure that the `pipeline-as-code-controller` is up by getting its route:
+2. Ensure that the `pipeline-as-code-controller` is up by getting its route. Remember to add this route as the webhook url of your Github App:
```
just an FYI: I didnt save the URL in the GitHub app yesterday and it seemed fine 🤔

thepetk replied:
Have you used an existing Github app? E.g you might have already configure your app to use the same webhook url.

maysunfaisal replied:
it had an older url which was a bit differemt

thepetk replied:
Yeah, I think is the Tekton repo CR that sets it up so it listens to the created repo. I'll revert this change.

thepetk replied:
Addressed here: [871c9bb](https://github.com/redhat-ai-dev/ai-lab-helm-charts/pull/20/commits/871c9bb423ede996799dc19d1b975017168cf5f3)

thepetk committed:
Revert note for webhook url

thepetk committed:
Update gitSourceRepo description

Jdubrick reviewed (approved):
lgtm after the changes

thepetk committed:
Fix values json description

thepetk committed:
Remove special chars from values.json

thepetk commented:
@maysunfaisal @Jdubrick I'll need a re-approval / review after one small change I've added. Thanks!
